src/ring_buffer.rs

```diff
@@ -68,18 +69,27 @@ impl<T: Copy + Default> RingBuffer<T> {
             self.head + self.capacity() - self.tail
         }
     }
-
+    /// Get the capacity of the ring buffer.
     pub fn capacity(&self) -> usize {
         // Return the size of the internal buffer.
         self.buffer.len()
     }
+
+
 }
 
 impl RingBuffer<f32> {
+    /// Return the value at an offset from the current read index.
     // Return the value at at an offset from the current read index.
     // To handle fractional offsets, linearly interpolate between adjacent values. 
     pub fn get_frac(&self, offset: f32) -> f32 {
-        todo!("implement")
+
```

Comment:
You may want to add an edge case statement for if offset = 0.0. It can behave a bit strange if the buffer is already full.